### ISSUE TITLE
Creates empty page ii depending on how often you run pdflatex

### ISSUE THREAD

**Issue by tbm** [fmarotta/kaobook#126]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Check that you are using the latest version of kaobook.
    * Check that your issue isn't already filed: https://github.com/fmarotta/kaobook/issues

### Description

When I run pdflatex once, I get a PDF with two pages: page i for the front page and page 1 for the chapter

If I run pdflatex one more time, the PDF has three pages: there's now a page ii which is complete empty.

If I now run pdflatex again, the PDF has two pages again.

Basically: running pdflatex an odd number of times: 2 pages; and even number of times: 3 pages

I'm not sure if I'm doing something wrong, or if this is a problem with the template.

Minimal example:

```
\documentclass{kaobook}
\usepackage[english]{babel}
\usepackage{blindtext}

\begin{document}

\title{Bug Report}
\author{Author}
\date{\today}

\frontmatter

\KOMAoptions{twoside=semi}
\maketitle
\KOMAoptions{twoside=false}

\mainmatter
\setchapterstyle{kao}

\chapter{First Chapter}
\blindtext

\end{document}
```

**Comment by AlexanderZeilmann** [comment]
This is a weird bug.

Here is what I found out:
* The `.aux` file changes between the runs.
* If I delete the `.aux` file and compile the document, it will always have two pages.
* The `.aux` file after 2, 4, 6, ... runs is identical.
* The `.aux` file after 3, 5, 7, ... runs is identical.
* The `.aux` file after 1 run is not the same as after 2 or 3 runs.
* The problem appears only in very minimal examples. Inserting a `\tableofcontents` just before `\mainmatter` fixes the bug.
* The bug appears with `pdflatex`, `LuaLaTeX` and `XeLaTeX`.

I am not familiar with reading `.aux` files, but perhaps the following is helpful to someone:

### Difference between `.aux` files after 1 and 2 runs
<img width="1802" alt="12" src="https://user-images.githubusercontent.com/901391/114355183-744fe400-9b6f-11eb-965f-cbab8e62dd65.png">

### Difference between `.aux` files after 2 and 3 runs
<img width="1802" alt="23" src="https://user-images.githubusercontent.com/901391/114355578-ede7d200-9b6f-11eb-99ec-3f4ec1916df5.png">

### Difference between `.aux` files after 1 and 3 runs
<img width="1806" alt="13" src="https://user-images.githubusercontent.com/901391/114355242-86318700-9b6f-11eb-8e42-77b11063bb76.png">

**Comment by tbm** [comment]
> The problem appears only in very minimal examples.

My document is around 60 pages so size doesn't seem to matter...

> Inserting a `\tableofcontents` just before `\mainmatter` fixes the bug.

... but this seems to be true.  After I added a TOC I haven't noticed this issue. (Now I always get the empty page, which I personally don't like, but which I guess is the expected behaviour.)

**Comment by AlexanderZeilmann** [comment]
>> The problem appears only in very minimal examples.

> My document is around 60 pages so size doesn't seem to matter...

Sorry, with "minimal examples", I didn't mean "short documents". I was referring to the fact, that once the book contains all parts a traditional printed book has, e.g.  \tableofcontents, the bug seems to go away.
The bug should, however, be fixed anyway.

> Now I always get the empty page, which I personally don't like, but which I guess is the expected behavior.

The empty page is a legacy from the time where books were still primarily printed and not read on screen.
I agree, that this empty page is superfluous nowadays.

**Comment by tbm** [comment]
@AlexanderZeilmann right, agreed. Thanks for posting your analysis!

**Comment by fmarotta** [comment]
The bug is caused by `\KOMAOptions{twoside=semi}`-`\KOMAOptions{twoside=false}`, which is a workaround I added in the old days because in one-side mode the `\uppertitleback` and `\lowertitleback` are not displayed. Currently I have removed the `\KOMAOptions` lines from my examples in GitHub, but I think they are still there in the Overleaf version of kaobook, and many users of course still have them in their documents. By removing the `\KOMAOptions` lines around `\maketitle`, the bug disappears, but `\uppertitleback` and `\lowertitleback` are not printed unless `twoside=true`.

If you want to remove the extra white page before the first chapter, simply pass `twoside=false` as a class option, and omit the `KOMAOptions` lines around `\maketitle`. Note that `twoside=false` is *not* the same tihng as `twoside=true` + `open=any`, because with the latter combination the extra white page is still there.

As general rules, I'd suggest to:
* Never use `\KOMAOptions`.
* Choose `twoside=true` or `twoside=false` according to your needs, but beware that with `twoside=false` you don't have the upper/lower titleback.
* If you don't want the extra whitepage in the frontmatter, choose `twoside=false`.
* If you want to have the upper/lower titleback, choose `twoside=true`.
* If you want to have the upper/lower titleback, and a one-side document at the same time, add manually the contents that you'd put in the upper/lower titleback. For instance, the copyright page in the example_and_documentation.pdf can be achieved by writing
```
...
\makeatletter
\@titlehead
\vfill
\textbf{Disclaimer}\\
You can edit this page to suit your needs. For instance, here we have a no copyright statement, a colophon and some other information. This page is based on the corresponding page of Ken Arroyo Ohori's thesis, with minimal changes.
    
\medskip

\textbf{No copyright}\\
\cczero\ This book is released into the public domain using the CC0 code. To the extent possible under law, I waive all copyright and related or neighbouring rights to this work.
    
To view a copy of the CC0 code, visit: \\\url{http://creativecommons.org/publicdomain/zero/1.0/}
    
\medskip
    
\textbf{Colophon} \\
This document was typeset with the help of \href{https://sourceforge.net/projects/koma-script/}{\KOMAScript} and \href{https://www.latex-project.org/}{\LaTeX} using the \href{https://github.com/fmarotta/kaobook/}{kaobook} class.
    
The source code of this book is available at:\\\url{https://github.com/fmarotta/kaobook}
    
(You are welcome to contribute!)
    
\medskip
    
\textbf{Publisher} \\
First printed in May 2019 by \@publishers
```

I think this behaviour is OK, but let me know :)

**Comment by tbm** [comment]
Sounds good. Maybe you can add the general rules to the docs, but anyway since you removed the KOMAoptions from the examples I think this can be closed.
